tokens/delivery/http/tokens_delivery.go

```diff
@@ -163,11 +172,23 @@ func (a *TokensHandler) GetPrices(c echo.Context) (err error) {
 		}
 	}
 
-	prices, err := a.TUsecase.GetPrices(ctx, baseDenoms, []string{defaultQuoteChainDenom}, domain.ChainPricingSourceType)
+	// Check if pricingSource param is valid
+	var pricingSource domain.PricingSourceType
+	pricingSourceParam := c.QueryParam("pricingSource")
+	if pricingSourceParam == "" {
+		pricingSource = domain.ChainPricingSourceType
+	} else {
+		pricingSourceInt, err := strconv.Atoi(pricingSourceParam)
+		if err != nil || !a.TUsecase.IsValidPricingSource(pricingSourceInt) {
+			return c.JSON(http.StatusBadRequest, domain.ResponseError{Message: fmt.Sprintf("invalid pricing source: %s", pricingSourceParam)})
+		}
+		pricingSource = domain.PricingSourceType(pricingSourceInt)
+	}
```

Comment:
```suggestion
	if pricingSourceParam == "" {
	         // By default, if no source is explicitly provided, we use a chain pricing source. 
		pricingSource = domain.ChainPricingSourceType
	} else {
	        // Otherwise, we attempt to parse the given pricing source.
		pricingSourceInt, err := strconv.Atoi(pricingSourceParam)
		if err != nil || !a.TUsecase.IsValidPricingSource(pricingSourceInt) {
			return c.JSON(http.StatusBadRequest, domain.ResponseError{Message: fmt.Sprintf("invalid pricing source: %s", pricingSourceParam)})
		}
		pricingSource = domain.PricingSourceType(pricingSourceInt)
	}
```

Reply:
Thanks for the comment suggestion.